src/txmempool.cpp

```diff
@@ -197,12 +197,28 @@ util::Result<CTxMemPool::setEntries> CTxMemPool::CalculateAncestorsAndCheckLimit
 }
 
 bool CTxMemPool::CheckPackageLimits(const Package& package,
+                                    const int64_t total_vsize,
                                     std::string &errString) const
 {
+    size_t pack_size = package.size();
+
+    // Package itself is busting mempool limits; should be rejected even if no staged_ancestors exist
+    if (pack_size > static_cast<uint64_t>(m_limits.ancestor_count)) {
+        errString = strprintf("package count %u exceeds ancestor count limit [limit: %u]", pack_size, m_limits.ancestor_count);
+        return false;
+    } else if (pack_size > static_cast<uint64_t>(m_limits.descendant_count)) {
+        errString = strprintf("package count %u exceeds descendant count limit [limit: %u]", pack_size, m_limits.descendant_count);
+        return false;
```

Comment:
Same here
```
diff --git a/src/txmempool.cpp b/src/txmempool.cpp
index 324292c055..a9d8355aa3 100644
--- a/src/txmempool.cpp
+++ b/src/txmempool.cpp
@@ -208,7 +208,7 @@ bool CTxMemPool::CheckPackageLimits(const Package& package,
         return false;
     } else if (pack_size > static_cast<uint64_t>(m_limits.descendant_count)) {
         errString = strprintf("package count %u exceeds descendant count limit [limit: %u]", pack_size, m_limits.descendant_count);
-        return false;
+        return true;
     } else if (total_vsize > m_limits.ancestor_size_vbytes) {
         errString = strprintf("package size %u exceeds ancestor size limit [limit: %u]", total_vsize, m_limits.ancestor_size_vbytes);
         return false;
```

Reply:
going to reserve for follow-up test writing